### PR TITLE
Regularization

### PR DESCRIPTION
**Pull request by dcoeurjo**
# PR Description

Digital Surface Regularization (DGCI2017)

Previously #299 

# Checklist

- [x] Doxygen documentation of the code completed (classes, methods, types, members...).
- [x] Main tool doxygen documentation (following existing documentation of [DGtalTools documentation](http://dgtal.org/doc/tools/nightly/).
- [x] Check if it follows the tools structure described in [CONTRIBUTING.md](https://github.com/DGtal-team/DGtalTools/blob/master/CONTRIBUTING.md)
- [x] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtalTools/blob/master/ChangeLog.md) added.
- [ ] Update the readme with potentially a screenshot of the tools if it applies. 
- [x] No warning raised in Debug ```cmake``` mode (otherwise, Travis C.I. will fail).

### CODE REVIEW

**Comment by dcoeurjo** [comment]
beside #307, this code can be reviewed

**Comment by kerautret** [comment]
nice, thanks, I will have a look once I found a moment ;)

**Comment by dcoeurjo** [comment]
;) 👌

File: volumetric/volCComponentCounter.cpp [hunk 1]
```diff
@@ -104,42 +104,40 @@ void CCCounter(Rank& r, Parent& p, const Image& elements, const unsigned int con
   typename Image::Point decz(0,0,1);
   
   //Merging process
-  for(typename Image::Domain::ConstIterator e = elements.domain().begin();
-      e !=elements.domain().end(); ++e)
+  for ( typename Image::Domain::ConstIterator e = elements.domain().begin();
```
[the remark this answers is not in the document]
Comment (dcoeurjo) [reply volumetric/volCComponentCounter.cpp]: not sure.. looks like formatting only

File: estimators/volSurfaceRegularization.cpp [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+{
+    namespace po = boost::program_options;
+
+    using DGtal::trace;
+    using std::endl;
+
+    Options options;
+    po::options_description po_shape("File options");
+    po_shape.add_options()
+        ("image-filename,i", po::value<std::string>(&options.image_filename)->default_value(""), "input vol filename for image shape or input cvs filename for surfels and normals")
```
[the remark this answers is not in the document]
Comment (dcoeurjo) [reply estimators/volSurfaceRegularization.cpp]: for the vol, the only assumption is that 0 is the background.

**Comment by kerautret** [comment]
fine, I think that we can merge, It works fine on set of full voxels:
capture d ecran 2018-03-16 a 10 00 22

**Comment by kerautret** [comment]
not tested on imported normals since the conversion with surfers need a little time to check in my tools. For the thin structure I put an issue.

**Pull request merged**